lib/Dialect/SPIRV/Serialization/Deserializer.cpp

```diff
@@ -581,7 +591,12 @@ LogicalResult Deserializer::processArrayType(ArrayRef<uint32_t> operands) {
            << defOp->getName();
   }
 
-  typeMap[operands[0]] = spirv::ArrayType::get(elementTy, count);
+  spirv::ArrayType::LayoutInfo layoutInfo = typeDecorations.lookup(operands[0]);
+  if (!layoutInfo) {
```

Comment:
This is not true. Only types used by interface variables need to be explicitly laid out. Types used by function local variables are not. So we should drop this check.

Reply:
Oh, sorry, my fault,
I was relying on this statement in the spec:

> **Each array type must have an ArrayStride decoration**, unless it is an array that contains a structure decorated with Block or BufferBlock, in which case it must not have an ArrayStride decoration.

and completely missed the header 

> Composite objects in the **StorageBuffer**, **Uniform**, and **PushConstant** Storage Classes must be explicitly laid out. The following apply to all the aggregate and matrix types describing such an object, recursively through their nested types:

Thanks a lot for fixing me!